### ISSUE TITLE
OWLdev.xlsx feature tabel actueel en nuttig maken en houden

### ISSUE THREAD

**Issue by LloydRutledge** [ABI-Team-30/Fresnel-Forms#43]
Ik het net https://svnext.ou.nl/INF_Studenten_ABI_team30/source/trunk/FresnelForms/design/OWFdev.xlsx op SVN gezet. Die is een tabel van features in OWL, Fresnel en SMW en de relaties tussen. Die "relaties" zijn eigen hoe dingen geïmplementeerd worden: OWL->Fresnel is de automatisch initiële Fresnel, en hoe die aangepast kan worden. Fresnel->SMW zijn features or feature requests for fresnel2wiki.

Kunnen jullie deze tabel bekijken om en zeggen of de tabel consequent is met wat jullie geïmplementeerd hebben? Zijn er fouten te corrigeren? Of voorstellen voor verhelding, uitbreiding of verbetering van de tabel? Hoor ik het graag! Of jullie kunnen de tabel zelf aanpassen.

Ik zet mezelf als assignee als eindverantwoordelijke dat de tabel deugt en correct is. Kan ieder van jullie verantwoordelijk zijn over of de tabel klopt met wat je zelf implementeert? En zo niet dan dat je of me vertelt of zelf de tabel op SVN aanpast?

De milestone is de Forms paper op 28 februari omdat ik een versie van deze tabel in de artikel zal zetten.

"Low hanging fruit" omdat het eist waarschijnlijk weinig werk. En "Must" omdat volgens mij hoort het om een sleutel rol te spellen in zowel implementatie als communicatie tussen ons als communicatie naar ons publiek.

En klein user story is dat zowel wij als de lezer van ons artikel deze tabel nuttig vind om de structuur van ons implementatie te begrijpen. Nog een is dat wij het nuttig vinden om het implementatie van nieuwe features te communiceren, plannen en prioriteren.

**Comment by LloydRutledge** [comment]
Ik heb die aangepast met kleur code statussen voor "al in PHP", al in Protégé plugin en met milestone Forms paper. Kunnen jullie die bekijken om te zien of het klopt?

**Comment by jheijning** [comment]
RDFS range value is still under development, but getting closer.

**Comment by LloydRutledge** [comment]
@jheijning, kan je me vertellen welke aangepast worden van kleur? Alles die geel is naar roze, of delen ervan. Of als met makkelijker is die zelf doen op het bestand op https://svnext.ou.nl/INF_Studenten_ABI_team30/source/trunk/FresnelForms/design/OWFdev.xlsx (en blijven doen). Of pas ik die zelf altijd aan als je verteld wat - wat makkelijker voor je is.

![range](https://cloud.githubusercontent.com/assets/1098321/6092605/538d4522-aee6-11e4-8c5c-20b4053ca9d7.jpg)

**Comment by jheijning** [comment]
Ja, alles naar naar roze, behalve (Value[:uri, :image, :externalLink], Fresnel), die naar geel. de datatype's is nu wel één van onze prioriteiten om te implementeren.

Liever dat jij ze aanpast voordat ik een fout maak. We zullen je op de hoogte houden.

**Comment by LloydRutledge** [comment]
Ik heb die naar roze gezet en ingecheckt. Maar veel van waren al in mijn PHP gedaan? Maar die was direct van de ontology naar SMW en niet via Fresnel met nieuwe OWF uitbreidingen. Ik neem aan dat dat de reden is waarom enkele al-in-PHP functies ineens op nieuwe geïmplementeerd moeten worden. Hoor ik het graag of deze (en welke dan ook) features geïmplementeerd worden in actueel gemaakt in deze tabel zijn of moeten.

**Comment by jheijning** [comment]
Ja, die waren in PHP gedaan, maar worden nu opnieuw geÏmplementeerd, omdat de SPARQL queries niet op fresnel uitgevoerd konden worden. Ik verwacht dat zondag's release van FForms deze functionaliteit weer zal hebben.

**Comment by jheijning** [comment]
De range values die geel waren in Lloyd's post met plaatje hierboven zijn weer geel en ook :showProperties rdf:List heb ik geel gemaakt. @LloydRutledge Kun je kijken of dit klopt? We kunnen met Fresnel Forms in de gui met drag en drop en dan in Fresnel met :showProperties rdf:List de properties op volgorde zetten in de wiki. zie plaatje:
![captura](https://cloud.githubusercontent.com/assets/9306134/6229205/7727c102-b6b5-11e4-8797-64c6d4f2ca0a.PNG)

**Comment by jheijning** [comment]
Ook @AlexMekkering  en @TeunTheunissen

**Comment by LloydRutledge** [comment]
Klopt. Ik heb die tabel nog wat aangepast. Geel in de SMW kolom is nu wit want die was al in PHP, hoewel van verschillende Fresnel-OWF - waar die nieuw is is nu geel. Klopt deze nu?

![naamloos](https://cloud.githubusercontent.com/assets/1098321/6254946/95818b2c-b7ac-11e4-8a6d-bf3b79898f04.png)

**Comment by LloydRutledge** [comment]
Deze tabel staat nu in de submission draft op https://svnext.ou.nl/INF_Studenten_ABI_team30/Environment%20&%20Misc/OWFdev.xlsx . Status kleuren in paper zijn alleen "implemented" en "planned". Dus de rood Forms milestone features zijn gewoon hier wit als implemented, en de priorities zijn met planned gemerged in de paper. Dus deze is wat we de wereld vertellen dat we nu al gedaan hebben of volgend week zeker doen. Nog om aan te passen natuurlijk. Ziet er nu zo uit:

![naamloos](https://cloud.githubusercontent.com/assets/1098321/6314317/7c02f282-b9da-11e4-90e8-24673c759ce2.png)

**Comment by AlexMekkering** [comment]
Added support for xsd:gYear and xsd:time in r393.
@TeunTheunissen, please review.

@LloydRutledge, I updated OWFdev.xlsx accordingly:
![image](https://cloud.githubusercontent.com/assets/9436555/6324906/e0b62020-bb40-11e4-9668-dc81d1c7cd38.png)

**Comment by LloydRutledge** [comment]
Thx!

**Comment by LloydRutledge** [comment]
Hoe wordt Geographic coordinate, Code, Temperature nu ondersteund in de plugin? Dat je in de Fresnel-OWF onto zitten, en vertaald worden door fresnel2wiki?

**Comment by TeunTheunissen** [comment]
Review r393 is OK

**Comment by jheijning** [comment]
Vanwege sluiting #45 heb ik de sheet aangepast:
![captura](https://cloud.githubusercontent.com/assets/9306134/6418940/5591c3f6-beb9-11e4-8c95-cd2a71fe1675.PNG)
De andere rode cellen in dezelfde groep begrijp ik niet helemaal.

**Comment by LloydRutledge** [comment]
Joop, ik denk dat de cellen links ook geel mogen zijn. Waar krijgt fresnel2wiki de gegevens voor [[Imported from::]] vandaan? Is het van de namespace en prefix gegevens in de bron ontology zelf?

**Comment by jheijning** [comment]
Uit de bronontologie (het jena model) inderdaad.

**Comment by LloydRutledge** [comment]
Tabel nu commit met deze onderdeel:

![naamloos](https://cloud.githubusercontent.com/assets/1098321/6426824/cc99e7b8-bf6c-11e4-93bd-32ea79caabd6.png)

**Comment by AlexMekkering** [comment]
I suggest to also label this issue with 'Post ABI-Team-30', just to be clear that this will be finalized after ABI Team 30 has finished their project.

**Comment by LloydRutledge** [comment]
Agreed, for now. I'll have to see how such a table works in a more open and spread collaboration.
